### DISCUSSION TITLE
Gothic2.exe  /  GothicStarter.exe / GothicMod.exe / Gothic.exe

### DISCUSSION THREAD

**Discussion by Kor-Angar** [dosinabox/gothic2_steam_fix#29] [General]
Вообщем не нашёл я Старетер который у меня был. Но можно сделать вот так

сделать копию Gothic2.exe и переименовать в GothicMod.exe
Gothic2.exe переименовываем в Gothic.exe 

создаём bat файл который будет совершать все необходимые манипуляции для стима. Вот черновой вариант моего батника
```
@ECHO OFF
cd system
rename Gothic2.exe KorAngar
start /wait GothicStarter.exe
rename KorAngar Gothic2.exe
```

> тут мы переименовываем файл Gothic2.exe, что бы стартер не ругался на него и запускаем стартер. а в конце опять переименовываем всё  обратно.

у меня всё работает. попробуй
![image](https://user-images.githubusercontent.com/49514764/86545548-4ed6df80-bf38-11ea-86f5-9c52f09eec03.png)

***

https://steamcommunity.com/sharedfiles/filedetails/?id=2155616084
[Steam_GothicStarter.zip](https://github.com/dosinabox/g2_steam_fix/files/4876162/Steam_GothicStarter.zip)

**Comment by Kor-Angar** [comment]
> В общем, вчера удалось собрать собственный стартер. Теперь всё будет как надо)

однако

**Comment by Kor-Angar** [comment]
ну вообщем то хорошо. стартер конечно лучше чем костыли с батниками :-)

**Comment by Kor-Angar** [comment]
[GraT  ](https://worldofplayers.ru/members/97625/)тоже делал и свой большой тяжелый стартер  и слегка тюнингавал Стартер пираней.

**Comment by dosinabox** [comment]
На основе его доработанного стартера (2.8) и сделан мой. Там изменение в одну строку, чтобы запускать нужный exe. Внешне ничего не поменялось.

**Comment by Kor-Angar** [comment]
Закидывай. Будем тестить 😀

**Comment by dosinabox** [comment]
На днях

**Comment by dosinabox** [comment]
Судя по версии в главном меню (2.6 (fix) без SP 1.8) и растянутому фону, сборник вообще не установился. Или ты установил поверх что-то, что сломало System Pack. Помимо всего прочего, он и интерфейс подстраивает под монитор (параметр Scale в system/SystemPack.ini).

**Comment by dosinabox** [comment]
Проверь что ты устанавливаешь и запускаешь. И не забывай про пункт широкоформатного монитора при установке сборника.

**Comment by Kor-Angar** [comment]
> Или ты установил поверх что-то, что сломало System Pack.

переустановил вручную СиСтемпак надпись появилась.

> Проверь что ты устанавливаешь и запускаешь. И не забывай про пункт широкоформатного монитора при установке сборника.

да. в этом было дело. удалил файл g2a_nr_scriptpatch_vХХ_autoscale и всё стало Ок. Я так понимаю что autoscale не нужен если стоит СистемПак
![image](https://user-images.githubusercontent.com/49514764/86530862-19dd7500-bec5-11ea-8c76-83dde544fe75.png)
![image](https://user-images.githubusercontent.com/49514764/86530865-24980a00-bec5-11ea-8535-1a95b579680b.png)

почему ты в сборку не хочешь добавить шрифты от Мухомора?
https://worldofplayers.ru/threads/25865/

**Comment by dosinabox** [comment]
> почему ты в сборку не хочешь добавить шрифты от Мухомора?

Вкусовщина. И пользователю нельзя будет показать разницу между старыми и новыми до установки новых.

**Comment by Kor-Angar** [comment]
при установке отказался от русской озвучки
![image](https://user-images.githubusercontent.com/49514764/86539590-70b76e80-bf06-11ea-8523-227f2b358f99.png)
в результате как я понял неофициальное обновление тоже не установилось.

**Comment by Kor-Angar** [comment]
переустановил. поставил две галки русификатор+неоф. обновление.
нормального стартера нет.
![image](https://user-images.githubusercontent.com/49514764/86539667-1965ce00-bf07-11ea-99ce-71dd6262a83b.png)
 в результате игра не запускается в оконном режиме без рамки.

**Comment by Kor-Angar** [comment]
> Вкусовщина

дело не в вкусовщине, а в том как отображаются шрифты при игре в высоком разрешение
![image](https://user-images.githubusercontent.com/49514764/86540402-c3485900-bf0d-11ea-8138-22d179cd0cc3.png)

![image](https://user-images.githubusercontent.com/49514764/86540404-c6434980-bf0d-11ea-8956-73a76a36b963.png)

**Comment by dosinabox** [comment]
> нормального стартера нет.

Что есть нормальный стартер? GothicStarter_mod.exe - только для модостроителей, поскольку использует отладчик, который превращает предупреждения в ошибки и открывает юзеру кучу окон во время игры. Обычный GothicStarter.exe в сборнике есть. Ни один из них пока нормально не работает из Стима.

> в результате как я понял неофициальное обновление тоже не установилось.

Если не выбирать русификацию текста, то устанавливается обновление без скриптов - исправленные модели, текстуры, анимации, миры, звуки. А иначе скрипты обновления сделают игру русской, чего пользователь не заказывал при установке, когда снял все галочки русификации. Для установки скриптов обновления требуется русификация текста.

Обе проблемы будут решены новым кастомным стартером.

**Comment by Kor-Angar** [comment]
всё сложно опять получается. тут галку ставить не надо, когда по логике надо. Потом зайти туда, удалить это, скопировать то.  Наверное это очень хорошая сборка для определенной аудитории. может быть даже и широкой аудитории.

**Comment by Kor-Angar** [comment]
широкоформатный монитор это что? в стиме у тебя  в топике вычитал что это какие то заставки.. не понятно :-(

**Comment by dosinabox** [comment]
> широкоформатный монитор это что?

Правильное отображение фона главного меню и загрузочных экранов:
![Screenshot_1](https://user-images.githubusercontent.com/39790055/86543611-eb918100-bf28-11ea-8b57-09d33ef535ca.jpg)
![Screenshot_2](https://user-images.githubusercontent.com/39790055/86543614-ecc2ae00-bf28-11ea-923e-0002f75dea54.jpg)

**Comment by Kor-Angar** [comment]
никогда бы не догадался :-)

мне кажется или у тебя шрифт лучше отображается?

**Comment by Kor-Angar** [comment]
> Если не выбирать русификацию текста, то устанавливается обновление без скриптов - исправленные модели, текстуры, анимации, миры, звуки. А иначе скрипты обновления сделают игру русской, чего пользователь не заказывал при установке, когда снял все галочки русификации. Для установки скриптов обновления требуется русификация текста.

опять же не понятно из инсталлера. я решил что речь идёт о русской озвучке.

**Comment by Kor-Angar** [comment]
> опять же не понятно из инсталлера. я решил что речь идёт о русской озвучке.

а если выбрать субтитры в меню, неофициальное обновление установится?

**Comment by Kor-Angar** [comment]
а почему русская озвучка Акеллы так мало весит? :-)
![image](https://user-images.githubusercontent.com/49514764/86545748-8c883800-bf39-11ea-9982-c2436cda6aed.png)

**Comment by dosinabox** [comment]
> а почему русская озвучка Акеллы так мало весит? :-)

формат ogg

> а если выбрать субтитры в меню, неофициальное обновление установится?

да

> мне кажется или у тебя шрифт лучше отображается?

Похоже на то. При том что у меня 4К монитор. Но почему так - не знаю.

**Comment by dosinabox** [comment]
> Обе проблемы будут решены новым кастомным стартером.

добавлен

**Comment by Kor-Angar** [comment]
а где его скачать можно?

**Comment by dosinabox** [comment]
https://github.com/dosinabox/g2_steam_fix/releases/tag/09.20

**Comment by Kor-Angar** [comment]
а сам мод стартер?

**Comment by dosinabox** [comment]
[g2_steam_starter.zip](https://github.com/dosinabox/g2_steam_fix/files/5251929/g2_steam_starter.zip)

**Comment by dosinabox** [comment]
Немного обновила твое руководство)

**Comment by Kor-Angar** [comment]
чудесно.
https://steamcommunity.com/sharedfiles/filedetails/?id=2155616084
 только ссылку с самого верха не стоило убирать. в последние время люди ссылки не могут найти :-) а гитхаб считает кол-во скачиваний?

**Comment by dosinabox** [comment]
> а гитхаб считает кол-во скачиваний?

нет

**Comment by Kor-Angar** [comment]
> нет

а яндекс считает ;-)

Добавил  ссылку на Стим Фикс
![image](https://user-images.githubusercontent.com/49514764/94346076-e5480800-0032-11eb-8193-d78f8ddd19ce.png)

**Comment by dosinabox** [comment]
> а яндекс считает ;-)

К сожалению, в Украине он заблокирован. Сразу отсекается немаленькая часть потенциальных потребителей.
У руководства есть статистика посещений, лайков и избранного)

**Comment by Kor-Angar** [comment]
подскажи пожалуйста а можно как то запустить GothicStarter_mod как то без плясок с переименованием нового Gothic2.exe?

**Comment by Kor-Angar** [comment]
хотелось бы что бы zSpy в соседном окошко что нибудь писал :-)

**Comment by dosinabox** [comment]
> хотелось бы что бы zSpy в соседном окошко что нибудь писал :-)

Практика очень хорошая. А покажи что у тебя в папке system сейчас.

**Comment by Kor-Angar** [comment]
> А покажи что у тебя в папке system сейчас.

не хотелось бы тебя ввергать в шок. 
![image](https://user-images.githubusercontent.com/49514764/140611350-c82d5ada-b60d-4dc9-88d6-742dadb59fe9.png)

сейчас для запуска я использую батник GothicStarter_mod
```
@ECHO OFF

rename Gothic2.exe KorAngar
start /wait GothicStarter_mod.exe
rename KorAngar Gothic2.exe
```

**Comment by dosinabox** [comment]
Воистину ужасает.
Этот GothicStarter_mod.exe будет использовать GothicMod.exe и переименования станут не нужны:
[GothicStarter_mod.zip](https://github.com/dosinabox/g2_steam_fix/files/7490557/GothicStarter_mod.zip)

**Comment by Kor-Angar** [comment]
отл. спасибо!

**Comment by dosinabox** [comment]
А у меня вот так:
<img width="467" alt="Screenshot_2" src="https://user-images.githubusercontent.com/39790055/140612082-a9b81570-87e0-41d7-a448-a2f6b3151d76.png">

**Comment by Kor-Angar** [comment]
я тестирую много. и мне как тестеру надо иметь горку дополнительных ini
![image](https://user-images.githubusercontent.com/49514764/140612132-2d5138db-7651-4988-ba15-bfe573476986.png)
что бы быстро запускать тот или иной мир и проверять то что мне нужно проверить :-)
```
[SETTINGS]
;Player=PC_HERO
Player=PC_TestHero
World=OldWorld\OldWorld.zen
;World=Addon\AddonWorld.zen
;World=NewWorld\NewWorld.zen
;World=NewWorld\DRAGONISLAND.zen
```

ввиду того что ум мой не острый нужно иметь копии ini файлов, что бы методом сравнения хоть что то  понимать
![image](https://user-images.githubusercontent.com/49514764/140612162-105b4044-582e-495c-945d-dd1fc91fe320.png)

подлый стим может пытаться обновить exe файл. это как тату которое об этом напоминает
![image](https://user-images.githubusercontent.com/49514764/140612205-417158da-3fdd-4219-8de4-26d706c5cd7f.png)

это напоминает мне о том что скрипты пишут люди, а мы люди не совершенны и склонны множить сущности
![image](https://user-images.githubusercontent.com/49514764/140612220-6a293021-92c0-4905-a97c-d5b9d93a1486.png)

это мне напоминает о том что если не хватает ума на решение сложной задачи, её следует разделить на несколько простых и решить по отдельности
![image](https://user-images.githubusercontent.com/49514764/140612283-82db599a-6eac-476e-8635-430ea737455c.png)

архивы это очень хорошо. их содержимое остаётся неизменным :-) когда их становится слишком много я их удаляю.
![image](https://user-images.githubusercontent.com/49514764/140612332-160ab093-5abb-4b0e-9ebd-c4022aff9feb.png)

**Comment by Kor-Angar** [comment]
> А у меня вот так:

спасибо. теперь я узнал что у меня есть один лишний exe файл  Gothic.exe

**Comment by dosinabox** [comment]
Издатель изменил исполняемый файл для запуска игры в Steam с Gothic2.exe на GothicStarter.exe. Теперь для запуска установленных вручную модов не нужны никакие костыли, а мой кастомный стартер больше не актуален и **работать не будет**.

В связи с этим прошу:
- удалить или скрыть руководство [Запуск модов в Gothic 2 через Steam](https://steamcommunity.com/sharedfiles/filedetails/?id=2155616084);
- добавить меня в авторы руководства [Steam Overlay and Mod Starter for Gothic 2](https://steamcommunity.com/sharedfiles/filedetails/?id=2240305342) для актуализации описания и ссылок;
- удалить раздел **как запустить Mod версию?** из руководства [Русская озвучка от Акеллы для Готика 2 в виде MOD](https://steamcommunity.com/sharedfiles/filedetails/?id=2251488739);
- убрать упоминание **Gothic 2 Steam Fix** в разделе **зачем мне озвучка в виде MOD?** в руководстве [Русская озвучка от Акеллы для Готика 2 в виде MOD](https://steamcommunity.com/sharedfiles/filedetails/?id=2251488739) (русская озвучка теперь доступна официально и в стим-фиксе её нет).

**Comment by Kor-Angar** [comment]
это скрыл https://steamcommunity.com/sharedfiles/filedetails/?id=2155616084

**Comment by Kor-Angar** [comment]
> добавить меня в авторы руководства [Steam Overlay and Mod Starter for Gothic 2](https://steamcommunity.com/sharedfiles/filedetails/?id=2240305342) для актуализации описания и ссылок;

добавил

**Comment by Kor-Angar** [comment]
> удалить раздел как запустить Mod версию? из руководства [Русская озвучка от Акеллы для Готика 2 в виде MOD](https://steamcommunity.com/sharedfiles/filedetails/?id=2251488739);

отредактировал:

 _Что бы запустить Mod версию необходимо иметь GothicStarter для Готики 2. Осенью 2022 издатель изменил исполняемый файл для запуска игры в Steam с Gothic2.exe на GothicStarter.exe и теперь дополнительно для запуска мода ничего устанавливать не надо._

**Comment by Kor-Angar** [comment]
> убрать упоминание Gothic 2 Steam Fix в 

убрал

**Comment by dosinabox** [comment]
> это скрыл https://steamcommunity.com/sharedfiles/filedetails/?id=2155616084

Не тот статус, поставь Hidden

**Comment by Kor-Angar** [comment]
доступ по ссылке. что не так то?
![image](https://user-images.githubusercontent.com/49514764/201522426-5bd2b8dd-8b87-4898-9493-34ece8133be7.png)

**Comment by dosinabox** [comment]
По ссылке это тоже хорошо. Спасибо!

**Comment by Kor-Angar** [comment]
>  По ссылке это тоже хорошо. Спасибо!

там просто надо добавить комментарий что дескать это больше неактуально для стим версии так как теперь всё прекрасно работает и со штатными исполнительными файлами.

**Comment by Kor-Angar** [comment]
> 
Начиная с июля 2022 года в сборник не включается русификация от Акеллы, так как она стала официально доступна в Steam-версии игры. Проверьте свойства игры в библиотеке, чтобы сменить её язык. Если по какой-то причине вам нужна русификация в составе Gothic 2 Steam Fix, то используйте архив версий на [GitHub проекта](https://steamcommunity.com/linkfilter/?url=https://github.com/dosinabox/g2_steam_fix/releases)[github.com].

картинку можно добавь для малограмотных https://steamcommunity.com/sharedfiles/filedetails/?id=757500304
![image](https://user-images.githubusercontent.com/49514764/201522783-e1a065dc-f6e3-4605-b697-db9fe0a51ff8.png)

**Comment by Kor-Angar** [comment]
оно?  http://www.gothic-library.ru/load/programms/gothic_ii_starter_source_code/2-1-0-179

**Comment by dosinabox** [comment]
исходники:
https://github.com/dosinabox/g2_steam_fix/tree/master/GothicStarter

> не могу понять как это работает

 а что это?

**Reply by Kor-Angar** [reply]
>> а как проект скомплиировать в exe файл ?

> открывается и компилируется в Visual Studio, но не помню какой версии
в целом интерфейс как у Gothic Sourcer

 я компилировал в прошлом году. исходники смотрел и c оф. сайта

**Reply by Kor-Angar** [reply]
что то там даже правил :-)

**Reply by Kor-Angar** [reply]
> исходники:

неоткрывается

**Reply by dosinabox** [reply]
Сейчас удалено, осталось в истории:
https://github.com/dosinabox/g2_steam_fix/tree/1350c4f8f40a9ab2afb5dffe50918b3016a6f421/GothicStarter

**Comment by Kor-Angar** [comment]
вот это  похоже на то
```
const LPCTSTR TXT_INVALID  = TEXT("!<invalid>");

    /* Validate */
    RtlSecureZeroMemory(&Buff, sizeof(Buff));
    GetPrivateProfileString(INI_SEC_FILE, INI_FILE_VDF, TXT_INVALID, Buff,
        MAX_INFO, FilePath);
    if (lstrcmp(Buff, TXT_INVALID) == 0)
        return FALSE;
```

**Reply by Kor-Angar** [reply]
```
[FILES]
VDF=!<invalid>
```

файл ини не будет отображаться в стартаре.

**Reply by Kor-Angar** [reply]
> вот это похоже на то

у нероманта некорректное описание этого параметра

**Comment by dosinabox** [comment]
> а как проект скомплиировать в exe файл ?

открывается и компилируется в Visual Studio, но не помню какой версии
в целом интерфейс как у Gothic Sourcer

**Comment by Kor-Angar** [comment]
@dosinabox  конвертируй в топик плиз

**Reply by Kor-Angar** [reply]
спс

**Comment by Kor-Angar** [comment]
```
[MOD]
BASE=g123
```

**Reply by Kor-Angar** [reply]
![image](https://github.com/dosinabox/g2_steam_fix/assets/49514764/8d062899-bb72-489b-a7f9-7fce1fc8b4ad)

```
    // название в ГотикМод.exe
	if (lstrcmpi(Info->name, Info->base))
	{
		lstrcat(Buff, TEXT(" -> "));
		lstrcat(Buff, Info->base);
		lstrcat(Buff, TEXT(".ini"));
	}

```

**Reply by Kor-Angar** [reply]
```
			/* base */
			if (lstrcmpi(info->name, info->base))
			{
				strcpyx(filename, g_AppDirectory, info->name, TEXT(".ini"), NULL);
				strcpyx(base_org, g_AppDirectory, info->base, TEXT(".ini"), NULL);
				if (!FileExists(base_org))
					base_tmp[0] = 0;
				else
				{
					if (!GetTempFileName(g_AppDirectory, TEXT("ini"), 0, base_tmp))
						lstrcpy(base_tmp, TEXT("ini12345.tmp"));
					MyMoveFile(base_org, base_tmp);
				}
				MyCopyFile(filename, base_org); 
			}
```
судя по коду это Алис и ни на что не влияет.

**Reply by Kor-Angar** [reply]
```
    /* Name */
    lstrcpyn(Info->name, FileName, lstrlen(FileName) - 3);  // убирает расширение?
    CharLower(Info->name);  // нижний регистр

	/* Alias */
    // читает lpAppName, lpKeyName, lpDefault, lpReturnedString, nSize, lpFileName
	if (!GetPrivateProfileString(INI_SEC_MODS, INI_MOD_BASE, Info->name, Info->base, MAX_PATH, FilePath))
		lstrcpy(Info->base, Info->name);
    CharLower(Info->base);  // нижний регистр
```
